pkg/kubelet/dockertools/kube_docker_client.go

```diff
@@ -177,9 +198,13 @@ func (d *kubeDockerClient) PullImage(image string, auth dockertypes.AuthConfig,
 	if err != nil {
 		return err
 	}
-	opts.ImageID = image
+	ctx, cancel := getDefaultContext()
+	defer cancel()
 	opts.RegistryAuth = base64Auth
-	resp, err := d.client.ImagePull(getDefaultContext(), opts, nil)
+	resp, err := d.client.ImagePull(ctx, image, opts)
```

Comment:
The image pulling is slow, but for now we only set timeout for the "connection construct" phase, which should be relatively fast.
After that we'll wait and periodically check the image pulling progress infinitely.